### PR TITLE
TS-4 FEATURE Add Positions and PlayerPositions Tables

### PR DESCRIPTION
**Pull request by dcarg**
WIP - see

Having an issue on the position seeds with key not being unique

Screenshot 2023-11-05 at 21 29 05

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -62,3 +64,27 @@ model TeamMember {
   @@map("teamMembers")
 }
 
+model Position {
+  id Int @id @default(autoincrement())
+  sport Sport @relation(fields: [sportId], references: [id])
+  sportId Int
+  key String @unique // position name i.e. 'prop'
+  numbers Int[] // jersey numbers for the position. I.e. 'prop' can be [1,3]
```
Comment (dcarg) [inline prisma/schema.prisma:72]: Could make this either a number || number[]

See the seeds for what I mean by this

File: prisma/seeds/positions.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      title: 'Outside Back',
+      category: 'back',
+      sportId: rugbyId,
+    },
+  ]
+
+  const records = positions.map(async position => (
+    await prisma.position.upsert({
+      where: { 
+        key: { key: position.key }
```
Comment (dcarg) [inline prisma/seeds/positions.ts]: Issue is here

Screenshot 2023-11-05 at 21 29 05

File: prisma/seeds/positions.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    },
+    {
+      key: 'centre',
+      numbers: [12,13],
+      title: 'Center',
+      category: 'back',
+      sportId: rugbyId,
+    },
+    {
+      key: 'outside_back',
```
Comment (Mark-Ball) [inline prisma/seeds/positions.ts]: camelcase pls

File: prisma/seeds/positions.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    {
+      key: 'hooker',
+      numbers: [2],
+      title: 'Hooker',
+      category: 'forward',
+      sportId: rugbyId,
+    },
+    {
+      key: 'lock',
+      numbers: [4,5],
```
Comment (Mark-Ball) [inline prisma/seeds/positions.ts]: lint these arrays so we have spaces after the `,`

File: prisma/schema.prisma [hunk 5]
```diff
@@ -62,3 +64,27 @@ model TeamMember {
...
+  title String // position name i.e. 'Prop'
+  category String // 'Forward' or 'Back'
+
+  PlayerPositions PlayerPosition[]
```
Comment (Mark-Ball) [inline prisma/schema.prisma]: not sure of the implications of this being pascalcase, but in our other models, these are camelcase

File: prisma/seeds/positions.ts (hunk 2 above)
Comment (Mark-Ball) [reply prisma/seeds/positions.ts]: you have an extra nested object here, should just be
```
where: { 
  key: position.key,
},
```

**Review by Mark-Ball — CHANGES_REQUESTED** [review]

**Comment by dcarg** [comment]
`yarn prisma db push`
`yarn prisma db seed`
`yarn prisma studio` # to see positions table

**Review by Mark-Ball — APPROVED** [review]

**Pull request merged**